<!-- -->

<launch>
	<!-- REQUIED by mk_interface -->
	<arg name="qcID" value="0" />
	<arg name="firmware" value="3908" />
	<arg name="serial_device1" value="usb-FTDI_FT232R_USB_UART_A900fRxi-if00-port0" />
	<arg name="serial_device2" value="usb-FTDI_FT232R_USB_UART_A900fQqb-if00-port0" />
	
	<!-- REQUIED by telekyb_core -->
	<arg name="rotortype" value="Quadcopter" />
	<arg name="takeOffHeight" value="-1.0" />
	<arg name="landHeight" value="-0.3" />
	<arg name="initialMass" value="1.572" />
	<arg name="emergencyThrust" value="60" />


	<!-- Joystick Logitech 510 Remapping -->
	<node name="Joystick_$(arg qcID)" pkg="telekyb_joystick" type="tJoy_node">
	      <param name="JoystickOptions/tDevicePath" value="/dev/input/js0" />
	      <param name="JoystickOptions/tPubName" value="joy_exp" />
	      <param name="JoystickOptions/tButtonRemapping" value="[4,1,2,3,6,5,7,8,0]" />

	      <param name="JoystickOptions/tAxesRemapping" value="[0,1,3,2,5,4,6,7]" />
	      <param name="JoystickOptions/tAxisMultiplier" value="[-1,1,-1,1,1, 1, 1, 1]" />
	</node>

	
<!-- telekyb core  -->
	<node name="TeleKybCore_$(arg qcID)" pkg="telekyb_core" type="telekyb_core_node">
		<param name="TeleKybCore/tRobotID" value="$(arg qcID)" />

		<param name="StateEstimatorController/tPublishRosTransform" value="false"/>
		
		<param name="StateEstimatorController/tPluginLookupName" value="state_estimators_plugin::EurocStateEstimator"/>
		<param name="EurocStateEstimator/tEurocStateEstimatorSeTopicName" value="/firefly/filteredState" />
		
		<param name="TrajectoryController/tPluginLookupName" value="trajectory_trackers_plugin::EurocTrajectoryTracker"/>
		<param name="EurocTrajectoryTracker/tCommandsTopic" value="/TeleKyb/$(arg qcID)/Commands" />
		<param name="EurocTrajectoryTracker/tTaskNumber" value="31" />
		
		<param name="PositionControl/tDerivGain" value="2.3" />
		<param name="PositionControl/tXDerivGain" value="1.0" />
		<param name="PositionControl/tYDerivGain" value="0.9" />
		<param name="PositionControl/tPropGain" value="3.5" />
		<param name="PositionControl/tXPropGain" value="0.65" />
		<param name="PositionControl/tYPropGain" value="0.8" />
		<param name="PositionControl/tIntegGain" value="2.2" />
		<param name="PositionControl/tXIntegGain" value="0.8" />
		<param name="PositionControl/tYIntegGain" value="0.8" />
		<param name="PositionControl/tSatIntTerm" value="1.0" />
		
		<param name="PositionControl/tZPropGain" value="25.5" />
		<param name="PositionControl/tZDerivGain" value="11.9" />
		<param name="PositionControl/tZIntegGain" value="9.0" />
		<param name="PositionControl/tGravity" value="9.8" />

		<param name="tBrakeInPosition" value="true" />

		<param name="PositionControl/tDerGainVelMode" value="4" />
		<param name="PositionControl/tIntegVelGain" value="2.0" />
		<param name="PositionControl/tSatIntVelTerm" value="4.0" />
		
		<param name="StandardMassEstim/tInitialMass" value="$(arg initialMass)"/>
		
		<param name="TrajectoryProcessorController/tTrajectoryModules" value='[]'/>
		<param name="tTakeOffDestination" value="[0.0, 0.0, $(arg takeOffHeight)]"/>
		<param name="tTakeOffVertically" value="false"/>
		<param name="tLandDestinationHeight" value="$(arg landHeight)" />
		<param name="tFlyToDestination" value="[1.0, 1.0, 0.0]"/>
	</node>
	
	

	<!-- Obstacle Provider -->
	<node name="Obstacle_$(arg qcID)" pkg="tk_obstacle" type="obstacle_provider">
	      <param name="ObstacleProviderController/tObstacleProviders" value='[ "tk_obstacle/SurroundingBox" ] '/>
	      <param name="ObstacleProviderController/tTKStateTopicName" value="/TeleKyb/TeleKybSystem_$(arg qcID)/Sensor/TKState" />
	      <param name="SurroundingBox/tMinSurrObsBox" value="[-7.0,-4.0,-3.0]"/>
	      <param name="SurroundingBox/tMaxSurrObsBox" value="[7.0,7.0,0.2]"/>
	</node>	

	<!-- Euroc Thing for the simulator -->
	<node name="euroc_solution" pkg="tk_gazebo_interface" type="tk_gazebo_interface">
 	       <param name="robotName" value="firefly" />
 	       <param name="robotID" value="$(arg qcID)" />
	</node>
	<node name="PoseKalmanFilter" pkg="tk_modelga" type="poseKalmanFilter">
		<param name="tInitialZCoordinate" value="0.08"/>
	</node>
	

	<!-- Experiment -->
	<node name="tk_exp_joystick" pkg="tk_exp_joystick" type="tk_exp_joystick">
	      <param name="ExperimentOptions/robotID" value="$(arg qcID)" />
	      <param name="ExperimentOptions/tJoystickTopic" value="/TeleKyb/tJoy/joy_exp" />
	      <param name="ExperimentOptions/tUseMKInterface" value="false" />
	</node>

</launch>
